tests/docker/Dockerfile

```diff
@@ -121,6 +118,14 @@ RUN mkdir -p /opt/scripts/consumer_offsets_recovery
 COPY --chown=0:0 tools/consumer_offsets_recovery /opt/scripts/consumer_offsets_recovery
 RUN python3 -m pip install --force --no-cache-dir -r  /opt/scripts/consumer_offsets_recovery/requirements.txt
 
+# Install our in-tree Java test clients / workloads
+COPY --chown=0:0 tests/java/kafka-verifier /opt/redpanda-tests/java/kafka-verifier
+COPY --chown=0:0 tests/java/compacted-log-verifier /opt/redpanda-tests/java/compacted-log-verifier
+COPY --chown=0:0 tests/java/tx-verifier /opt/redpanda-tests/java/tx-verifier
+RUN mvn clean package --batch-mode --file /opt/redpanda-tests/java/kafka-verifier --define buildDir=/opt/kafka-verifier/
+RUN mvn clean package --batch-mode --file /opt/redpanda-tests/java/compacted-log-verifier --define buildDir=/opt/compacted-log-verifier
+RUN mvn clean package --batch-mode --file /opt/redpanda-tests/java/tx-verifier --define buildDir=/opt/tx-verifier
```

Comment:
in my experience those Java builds takes the longest to rebuild, having them at the end of docker file will make any other change in the docker file triggering rebuild of those java dependencies.

Reply:
I don't have a lot of Java background -- for the third party Java stuff I assume the slow builds are unavoidable, but for our own tools can we do something to make them build faster?  It seems really surprising how long it is taking to build small test workloads, makes me think it should be fixable.

Reply:
PR removes kafka-verifier & compacted-log-verifier and the new workloads piggybacks on tx-verifier so the overall build time has reduced. The clean rebuild of all verifies takes 20s

```
real    0m21.085s
user    0m20.034s
sys     0m1.351s
```

Reply:
then perhaps we can do the dependency fetching at the top of the dockerfile and the faster moving bits at the end?

Reply:
I'm reassured that the verifier build is <20s -- the other problematic deps had been much longer than that, and this is in line with how long other small-ish tools are taking to build, so don't think we have a problem here.

At some point we should rearrange things a little bit so that everything that's truly fast (like, stuff that doesn't compile code) comes last, but IMO we can save that for another day.